common/src/main/java/com/google/auto/common/MoreTypes.java

```diff
@@ -205,9 +243,40 @@ private static boolean equal(TypeMirror a, TypeMirror b, Set<ComparedElements> v
     EqualVisitorParam p = new EqualVisitorParam();
     p.type = b;
     p.visiting = visiting;
+    if (INTERSECTION_TYPE != null) {
+      if (INTERSECTION_TYPE.isInstance(a)) {
```

Comment:
On JavaWriter @tbroyer pointed out that the docs for `TypeMirror` discourage this.

> To implement operations based on the class of an TypeMirror object, either use a visitor or use the result of the getKind() method. Using instanceof is not necessarily a reliable idiom for determining the effective class of an object in this modeling hierarchy since an implementation may choose to have a single object implement multiple TypeMirror subinterfaces.